core/src/main/java/com/netflix/conductor/core/execution/AsyncSystemTaskExecutor.java

```diff
@@ -126,14 +129,19 @@ public void execute(WorkflowSystemTask systemTask, String taskId) {
                     task.getTaskId(),
                     task.getStatus());
 
-            boolean isTaskAsyncComplete = systemTask.isAsyncComplete(task);
             if (task.getStatus() == TaskModel.Status.SCHEDULED || !isTaskAsyncComplete) {
                 task.incrementPollCount();
             }
 
             if (task.getStatus() == TaskModel.Status.SCHEDULED) {
                 task.setStartTime(System.currentTimeMillis());
                 Monitors.recordQueueWaitTime(task.getTaskDefName(), task.getQueueWaitTime());
+
+                if (task.getTaskType().equals("KAFKA_PUBLISH") && isTaskAsyncComplete) {
```

Comment:
Immediately set a kafka publish task to `IN_PROGRESS`. 

In a high throughput scenario, the message was published and consumed again before this task was updated, causing a race condition where a completed task is later set as in progress.